### PR TITLE
*: show an unsynced message for workers which didn't get any shard DDL

### PR DESCRIPTION
**Pull request by lance6716**
### What problem does this PR solve? 
some users don't use `show-ddl-locks` to get unsynced tables, they use `query-status` instead, so we also show reasonable message in `query-status`

### What is changed and how it works?
- re-fill the related field
- also fix a BUG that forget to decrypt password before open connection to upstream which introduced in https://github.com/pingcap/dm/pull/1619

### Check List 

Tests 

 - Integration test

Code changes

 - Has (user) interface methods change

Related changes

 - Need to cherry-pick to the release branch
 - Need to be included in the release note

### CODE REVIEW

**Review by Ehco1996 — APPROVED** [review]

**Review by lichunzhu — COMMENTED** [review]
/lgtm

File: dm/master/server.go [hunk 1]
```diff
@@ -691,6 +691,45 @@ func (s *Server) QueryStatus(ctx context.Context, req *pb.QueryStatusListRequest
...
+				if _, ok := found[target]; ok {
+					continue
+				}
+				syncStatus.UnresolvedGroups = append(syncStatus.UnresolvedGroups, &pb.ShardingGroup{
+					Target:   target,
+					Unsynced: []string{"this DM-worker doesn't receive any shard DDL of this group"},
+				})
+			}
+		}
+	}
```
Comment (lichunzhu) [inline dm/master/server.go:731]: Weak advice: can we use a separate function for this part and add a unit test for it?

Comment (lance6716) [reply dm/master/server.go:751]: https://github.com/pingcap/dm/pull/1638/commits/31d5d1a4629fc1087f0760b6dd599b0e6a6382ca

**Comment by lance6716** [comment]
/hold 

If user only queries one source not whole task, current code is not awared of unsynced workers. So I will change the implementation to use `ShowLocks` of pessimist/optimist

**Comment by GMHDBJD** [comment]
any test for optimistic mode?

**Comment by lance6716** [comment]
> any test for optimistic mode?

seems optimistic mode will discover sources in a shard group lazily, so a totally unsynchronized DM-worker is not counted into shard group. will check later

**Comment by lance6716** [comment]
Oh currently query-status doesn't show `unresolvedGroups` for optimistic tasks. I'll add a TODO

https://github.com/pingcap/dm/blob/423c81d439d3f7ec43f8e72daa8ed21f727b53b1/syncer/status.go#L51

**Review by GMHDBJD — APPROVED** [review]
/lgtm

**Pull request merged**